pygmt/base_plotting.py

```diff
@@ -986,28 +984,16 @@ def text(
         if position is not None and isinstance(position, str):
             kwargs["F"] += f'+c{position}+t"{text}"'
 
-        with GMTTempFile(suffix=".txt") as tmpfile:
-            with Session() as lib:
-                fname = textfiles if kind == "file" else ""
-                if kind == "vectors":
-                    if position is not None:
-                        fname = ""
-                    else:
-                        pd.DataFrame.from_dict(
-                            {
-                                "x": np.atleast_1d(x),
-                                "y": np.atleast_1d(y),
-                                "text": np.atleast_1d(text),
-                            }
-                        ).to_csv(
-                            tmpfile.name,
-                            sep="\t",
-                            header=False,
-                            index=False,
-                            quoting=csv.QUOTE_NONE,
-                        )
-                    fname = tmpfile.name
-
+        with Session() as lib:
+            file_context = dummy_context(textfiles) if kind == "file" else ""
+            if kind == "vectors":
+                if position is not None:
+                    file_context = dummy_context("")
+                else:
+                    file_context = lib.virtualfile_from_vectors(
+                        np.atleast_1d(x), np.atleast_1d(y), np.atleast_1d(text)
+                    )
```

Comment:
Is it more readable to rewrite the codes to:
```python
if kind == "file":
   ...
elif kind == "vectors":
  ...
else:
  ...
```

Reply:
Yes it would be more readable. I think the only reason I wrote it that way was because pylint complained about too many if-statements:

```python
R0912: Too many branches (13/12) (too-many-branches)
```

If readability is more important, I can increase the branch limit on our .pylintrc file, or I can try refactoring more of the code in `text` to reduce the number of if-statements.

Reply:
> OR, we can just merge this in now and worry about it in the next PR that _will_ fix #483.

OK to me.